### PR TITLE
Issue a certificate move workload to top

### PR DESCRIPTION
**Pull request by meyskens**
This adds a first step where we issue a demo cert and look into how cert-manager handles it first, then we do the workload then the ingress.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,90 +113,117 @@ Now we can apply this configuration to the cluster using:
...
+    name: venafi-tpp-issuer
+    kind: Issuer
 ```
 
-After changing and saving the file we can deploy it using `kubectl apply -f hello-world.yaml`
-
-Now we have an Ingress deployed with a `cert-manager.io/issuer` annotation, this will tell cert-manager to use a specific issuer to automatically fetch and assign a certificate according to the policy you have configured on your Issuer resource.
-We can see them being issued using:
+Here we see a certificate resource for `demo.example.com` issued by the `venafi-tpp-issuer` we created before.
+We can nog add it to out cluster using:
```
Comment (munnerz) [inline README.md]: `nog` -> `now`
`out` -> `our`

File: README.md [hunk 2]
```diff
@@ -113,90 +113,117 @@ Now we can apply this configuration to the cluster using:
...
-*  start date: Feb 20 14:51:12 2020 GMT
-*  expire date: Feb 19 14:51:12 2021 GMT
-*  issuer: DC=local; DC=traininglab; CN=traininglab-Root-CA
-*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
+$ kubectl get certificates
+NAME               READY   SECRET     AGE
+demo-certificate   True    demo-tls   10s
 ```
-Notice that the issue in this example is being signed by a training Venafi TPP instance which is not a trusted certificate authority.
+When creating a `Certificate` resource cert-manager will generate a private key and a Certificate Signing Request. The private key is stored inside the secret we defined in teh configuration: `demo-tls`. This is stored in the Kubernetes secret store.
```
Comment (munnerz) [inline README.md]: nit: comma after `resource`

Comment (munnerz) [reply README.md]: Also `teh` -> `the`

Also maybe write `secret` as `Secret resource` to make it clear?

File: README.md [hunk 3]
```diff
@@ -113,90 +113,117 @@ Now we can apply this configuration to the cluster using:
...
+The Certificate Signing Request (CSR) is stored inside a newly generated `CertificateRequest` resource.
+```console
+$ kubectl get certificaterequests
+NAME                          READY   AGE
+demo-certificate-4260521829   True    10s
+```
+To get more info about `CertificateRequest` resource, including the event history and any errors we can use `kubectl describe`:
```
Comment (munnerz) [inline README.md]: Maybe link to the 'concepts' page in our docs about CSRs?

File: README.md [hunk 4]
```diff
@@ -113,90 +113,117 @@ Now we can apply this configuration to the cluster using:
...
+Here we see our generated CSR and Certificate (both base64 encoded), as well as detailed info of the status of the CertificateRequest.
 
-#### Using Chrome
-Open your browser and browse to the hostname of your instance.
-You will see an "insecure" warning, this is because for this lab we used a training certificate authority which is not trusted by our computer.
-![Chrome warning](./images/chrome-error.png)
-Type "this is unsafe" on your keyboard (don't worry you won't see any letters appear), this will bypass the warning screen and you will be presented with our "Hello World" service.
-In the top bar click on "Not Secure" then click on "Certificate", here you can see the info about the certificate which we just issued.
-![Chrome certificate details](./images/chrome-cert.png)
+Once we see here that our certificate is issued by the Venafi TPP instance we can do the same describe on our secret resource that we asked cert-manager to put the certificate into:
```
Comment (munnerz) [inline README.md]: `describe on our secret ...` -> `describe on the Secret ...`

('the' and 'Secret')

File: README.md [hunk 5]
```diff
@@ -365,3 +392,85 @@ $ curl https://localhost:4430 -k -v
...
+```
+The private key and certificate have been saved into `venafi-demo-tls` inside the Kubernetes secret store from where NGINX picks it up.
+The NGINX Ingress controller has native support for reading TLS secrets from Kubernetes.
+
+### Testing the deployment
+
+#### Using curl
+You can see it being served using:
+```console
+$ curl https://<hostname> -k -v
```
Comment (wallrj) [inline README.md]: It seems a bit weird to have the flags after the URL. Consider putting the URL last.

File: README.md [hunk 6]
```diff
@@ -365,3 +392,85 @@ $ curl https://localhost:4430 -k -v
...
+* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
+* ALPN, server accepted to use http/1.1
+* Server certificate:
+*  subject: O=cert-manager
+*  start date: Feb 20 14:51:12 2020 GMT
+*  expire date: Feb 19 14:51:12 2021 GMT
+*  issuer: DC=local; DC=traininglab; CN=traininglab-Root-CA
+*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
+```
+Notice that the issue in this example is being signed by a training Venafi TPP instance which is not a trusted certificate authority.
```
Comment (wallrj) [inline README.md]: typo issue > issuer

File: certificate.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  name: demo-certificate
+  namespace: default
+spec:
+  secretName: demo-tls
+  duration: 2160h # 90d
+  dnsNames:
+    - demo.example.com
+  issuerRef:
+    name: venafi-tpp-issuer
+    kind: Issuer
```
Comment (wallrj) [inline certificate.yaml]: Add a new line.

**Pull request merged**